### ISSUE TITLE
Why  the  criterion_GAN(line 50)  in pix2pix.py  using MSELoss but not BCELoss?

### ISSUE THREAD

**Issue by wywd** [eriklindernoren/PyTorch-GAN#102]
# Loss functions
criterion_GAN = torch.nn.MSELoss()
criterion_pixelwise = torch.nn.L1Loss()

**Comment by nixczhou** [comment]
Well, I think you can use BCELoss() too, but you need to add sigmoid layer in the discriminator to let the output in range 0 to 1

**Comment by AnimationFan** [comment]
I also felt comfused today. So I've check the origin code of pix2pix. I found that both loss functional are avalible in pix2pix origin code. And my partner infom me that's use the advise of WGAN， which used MSE  loss replace softmax function. Hope it can help you.

> # Loss functions
> criterion_GAN = torch.nn.MSELoss() criterion_pixelwise = torch.nn.L1Loss()

**Comment by baicaiPCX** [comment]
您好，您的邮件小皮已经收到啦！！！

**Comment by lanlanwei** [comment]
您好，您的来信我已收到，感谢来信，祝天天好心情。
